libbeat/common/version.go

```diff
@@ -69,6 +69,10 @@ func NewVersion(version string) (*Version, error) {
 	return &v, nil
 }
 
+func (v *Version) IsValid() bool {
+	return v.version != ""
```

Comment:
Should this be called IsEmpty instead?

Reply:
Don't care about the name. stdlib seems to `IsZero() bool` for some types.

Reply:
Keeping IsValid cause of docstring:

`IsValid returns true if the version object stores a successfully parsed version number.`

The NewVersion constructor is no simple constructor, but actually a parser. Wonder why we parse ourselves. [github.com/hashicorp/go-version](https://github.com/hashicorp/go-version) is a pretty good and flexible versions package: